superset/views/core.py

```diff
@@ -1198,37 +1200,91 @@ def recent_activity(  # pylint: disable=no-self-use
             limit = int(request.args["limit"])
         else:
             limit = 1000
+        actions = request.args.get("actions", "explore,dashboard").split(",")
+        # whether to get distinct subjects
+        distinct = request.args.get("distinct") != "false"
 
-        qry = (
-            db.session.query(Log, Dashboard, Slice)
-            .outerjoin(Dashboard, Dashboard.id == Log.dashboard_id)
-            .outerjoin(Slice, Slice.id == Log.slice_id)
-            .filter(
-                and_(
-                    Log.action.in_(("queries", "shortner", "sql_json")),
-                    Log.user_id == user_id,
+        has_subject_title = or_(
+            and_(
+                Dashboard.dashboard_title is not None, Dashboard.dashboard_title != "",
+            ),
+            and_(Slice.slice_name is not None, Slice.slice_name != ""),
+        )
+
+        if distinct:
+            one_year_ago = datetime.today() - timedelta(days=365)
+            subqry = (
+                db.session.query(
+                    Log.dashboard_id,
+                    Log.slice_id,
+                    Log.action,
+                    func.max(Log.dttm).label("dttm"),
                 )
+                .group_by(Log.dashboard_id, Log.slice_id, Log.action)
+                .filter(
+                    and_(
+                        Log.action.in_(actions),
+                        Log.user_id == user_id,
+                        # limit to one year of data to improve performance
+                        Log.dttm > one_year_ago,
+                        or_(Log.dashboard_id is not None, Log.slice_id is not None),
```

Comment:
`is not None` will be evaluated prior to compilation, should be replaced by
```suggestion
                        or_(Log.dashboard_id != None, Log.slice_id != None),
```

Reply:
Thanks! I think changed this due to a pylint rule. No wonder I'm still getting null dashboard ids and had to use title checks later.

Reply:
Yeah, it's a shame the linter can't (to my knowledge) be set to ignore those checks on sqla orm ops.

Reply:
I changed to use `.isnot(None)`.

Reply:
Oh nice, I'll start using this, too 👍